This patch set fixes a bunch warnings

fidencio opened the pull request:
Those patches were part of PR#50, which got stuck for a long time and ended up closed as it didn't seem interesting for the project.

However, after some time another PR (PR#371) fixes the very same issues. So, I'd like to have my patches (at least) considered and, if possible, reviewed.

In any case, this PR is basically PR#50 without the patch that makes use of gnulib many warnings.

fidencio commented:
retest this, please

lslebodn commented on src/providers/ad/ad_gpo.c:
```diff
@@ -100,7 +100,6 @@
 /* If INI_PARSE_IGNORE_NON_KVP is not defined, use 0 (no effect) */
 #ifndef INI_PARSE_IGNORE_NON_KVP
 #define INI_PARSE_IGNORE_NON_KVP 0
-#warning INI_PARSE_IGNORE_NON_KVP not defined.
```
I am not sure whether it is the latest version of patch set from #50 because this warning is intentionally here.
And was removed from #50

https://github.com/SSSD/sssd/pull/50#discussion_r87219308

lslebodn commented on src/providers/krb5/krb5_child.c:
```diff
@@ -1981,7 +1981,7 @@ static errno_t tgt_req_child(struct krb5_req *kr)
                                         password_or_responder(password),
                                         sss_krb5_prompter, kr, 0,
-                                        SSSD_KRB5_CHANGEPW_PRINCIPAL,
+                                        discard_const(SSSD_KRB5_CHANGEPW_PRINCIPAL),
```
It was agreen in #50 that this patch will not be included
https://github.com/SSSD/sssd/pull/50#discussion_r87217214

lslebodn commented on src/tests/sbus_codegen_tests.c:
```diff
@@ -115,15 +115,15 @@ END_TEST
 START_TEST(test_signals)
 {
-    const struct sbus_signal_meta *signal;
+    const struct sbus_signal_meta *sm_signal;
```
I would prefer `ssm_singal` (Sbus Signal Meta) or `sgnl` `sig`

lslebodn replied:
And you kind of agreed in https://github.com/SSSD/sssd/pull/50#discussion_r87221861

lslebodn commented on src/tools/sssctl/sssctl_cache.c:
```diff
@@ -31,10 +31,10 @@
+#define SSSCTL_CACHE_CREATE {_("Cache entry creation date"), SYSDB_CREATE_TIME, attr_time_fn}
+#define SSSCTL_CACHE_UPDATE {_("Cache entry last update time"), SYSDB_LAST_UPDATE, attr_time_fn}
+#define SSSCTL_CACHE_EXPIRE {_("Cache entry expiration time"), SYSDB_CACHE_EXPIRE, attr_expire_fn}
+#define SSSCTL_CACHE_IFP    {_("Cached in InfoPipe"), SYSDB_IFP_CACHED, attr_yesno_fn}
```
Please check comment https://github.com/SSSD/sssd/pull/50#discussion_r87236342

lslebodn commented:
There are some patches with initialize some variables to NULL, 0 .. on rhle6
I assume they are caused by old gcc in e6 + combination with `-O1`. I was enabled due to coverage bug in el6 057f699b401097ad7f48fe3760e81c6993619a4b. I would prefer to drop them

I'm fine with suppressing warnings on el6 with `-02`

lslebodn commented:
Thank you for reviving this patchset with fixed warnings

fidencio commented:
@lslebodn, for some reason when I reset the patchset I didn't have the latest version of the patches :-/

Okay, now I have it and will re-work according to your comments.
May I add your patches in PR #371 here as well? (of course, keeping your authorship)

fidencio commented:
I have updated the patch set, now using the correct version of the patches done for PR #50. Also, I have added to this PR one patch from PR #371 that was not part of PR #50.

IMHO, PR #371 should be closed and this one should be reviewed instead.

lslebodn commented on src/tests/cmocka/test_sdap_initgr.c:
```diff
@@ -195,7 +195,15 @@ struct sdap_get_initgr_state *prepare_state(struct test_sdap_initgr_ctx *ctx,
     }
     assert_non_null(state->dom);
     assert_non_null(recent_dom_info);
-    recent_dom_info->next = NULL;
```
I could not see such warning in #371 and in was not part of previous patchset.
You might hide some bugs in test. Therefore NACK to this patch.

Rest of patch set is good and can be pushed

fidencio replied:
Okay, this patch has been dropped from this PR.

fidencio commented:
Patch set has been updated addressing @lslebodn's comments.

Thanks for the review.

lslebodn commented:
On (25/09/17 12:54), fidencio wrote:
>fidencio commented on this pull request.
>
>
>
>> @@ -195,7 +195,15 @@ struct sdap_get_initgr_state *prepare_state(struct test_sdap_initgr_ctx *ctx,
>     }
>     assert_non_null(state->dom);
>     assert_non_null(recent_dom_info);
>-    recent_dom_info->next = NULL;
>
>Okay, this patch has been dropped from this PR.
>

Do you have an idea why you can see such warning?
but others cannot

LS

lslebodn commented:
ACK

fidencio commented:
@lslebodn, IIRC I saw that in our CI but not on my system.

So, in case I face it again, I'll add it as part of the PR #378.

lslebodn commented:
retest this please

lslebodn commented:
master:
* 39e300314cb43e534179a6950274e1e9c9a48465
* d8d49ae91708a2360c5c8f7a01e531952e2f7771
* 01f852fcb27e0a0a07e2b8b51e7e838a256b485f
* 3cb4592e27dac8032bd528b6a9f1b49dfa4ec0b8
* 1ef1056e73e306eb042c4d58f9cfaee621a5cbbe
* 47863dee64fa93859bc16ddc387010734da665ee
* 76ff0b1dd08b4f7320daa2d63ed4f052757b8ddf
* 1a9cdc6ba3d1f9101d84992f6e37e8739e6a9d7c
* 0e6248c60791ccfe840a2938f27b56b9e72370e3
* f194bd0fa75f87ac29dc08c3aeca9aa86311bfc5
* 58a9b4f0b195f6ea1271d24990bf4df04eccd6ed
